File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  for (const index1 in nums) {
    for (const index2 in nums) {
      if (index1 < index2) {
        if (nums[index1] + nums[index2] === target) {
          return [index1, index2];
        }
      }
    }
  }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        chached = {}

        for nums_index in range(len(nums)):
            if nums[nums_index] in chached:
                return [chached[nums[nums_index]][1], nums_index]
            else:
                chached[target - nums[nums_index]] = [nums[nums_index], nums_index]